File: w6/echon.py
#!/usr/bin/python
import sys,re
if len(sys.argv) != 3:
   sys.stderr.write("Usage: %s <number of lines> <string>\n" % sys.argv[0])
   sys.exit(1)
number = re.match(r'[0-9]', sys.argv[1])
if number == None:
   sys.stderr.write("Usage: %s <number of lines> <string>\n" % sys.argv[0])
   sys.exit(1)
char = re.match(r'[a-zA-Z]', sys.argv[2])
if char == None:
   sys.stderr.write("Usage: %s <number of lines> <string>\n" % sys.argv[0])
   sys.exit(1)
value = int(sys.argv[1])
for i in range(0,value):
    print sys.argv[2]

File: w5/echon.pl
#!/usr/bin/perl -w
$n = $ARGV[0] ;
$string = $ARGV[1] ;
$number = $#ARGV + 1 ;
if ($n !~ /^\d+$/ || $number > 2 || $number < 2 || $string !~ /^[a-zA-Z]+$/) {
  die "Usage: ./echon.pl <number of lines> <string> at ./echon.pl line 3. $!\n" ;
}
$counter = 0 ;
while ($counter < $n) {
  print $string . "\n" ;
  $counter++ ;
}
